Sources/Swiftfall/Swiftfall.swift

```diff
@@ -1,635 +1,637 @@
-        public let loyalty: String?
-        
-        // This card’s colors.
-        public let colors:[String]?
-        
-        // Online listings for these cards names.
-        public let purchase_uris: [String:String]
-    
-        // Flavor text on the card, if there is any
-        public let flavor_text: String?
-        
-        // id of the illustration
-        public let illustration_id: String?
-        
-        // uris of the images
-        public let image_uris:[String:String]?
-        
-        public func simplePrint(){
-            // if the card has multiple faces, print them
-            if (self.card_faces) != nil {
-                for face in card_faces! {
-                    face.simplePrint()
-                }
-                return
-            } 
-            
-            // Each variable is tested to see if printing it makes sense.
-            if self.name != nil {
-                print("Name: \(name!)")
-            }
-            if self.mana_cost != nil {
-                print("Cost: \(mana_cost!)")
-            }
-            if self.type_line != nil {
-                print("Type Line: \(type_line!)")
-            }
-            if self.oracle_text != nil {
-                print("Oracle Text:\n\(oracle_text!)")
-            }
-            if self.power != nil && self.toughness != nil {
-                print("Power: \(power!)\nToughness: \(toughness!)")
-            }
-            if self.loyalty != nil {
-                print("Loyalty: \(loyalty!)")
-            }
-            
-        }
-    }
-    
-    // fuzzy
-    public static func getCard(fuzzy: String) -> Card?
-    {
-        let encodeFuzz = fuzzy.addingPercentEncoding(withAllowedCharacters: .urlHostAllowed)!
-        let call = "cards/named?fuzzy=\(encodeFuzz)"
-
-        var card: Card?
-        var stop = false
-        parseCard(call: call){
-            (newcard:Card?) in
-            card = newcard
-            stop = true
-        }
-        
-        while(!stop){
-            // Do this until parseCard is done
-        }
-        
-        return card
-    }
-    
-    
-    // exact
-    public static func getCard(exact: String) -> Card?
-    {
-        let encodeExactly = exact.addingPercentEncoding(withAllowedCharacters: .urlHostAllowed)!
-        let call = "cards/named?exact=\(encodeExactly)"
-        
-        var card: Card?
-        var stop = false
-        parseCard(call: call){
-            (newcard:Card?) in
-            card = newcard
-            stop = true
-        }
-        
-        while(!stop){
-            //Do this until parseCard is done
-        }
-        
-        return card
-    }
-    
-    // fuzzy
-    public static func getRandomCard() -> Card?
-    {
-        let call = "cards/random"
-        
-        var card: Card?
-        var stop = false
-        parseCard(call: call){
-            (newcard:Card?) in
```

Comment:
Same as above.

Reply:
Also, this code is repeated more than 2 times. That's usually an indicator that there should be some abstraction in place to avoid repeating this code. You probably want a function like `synchronouslyGetCard(call: String)` that will perform the semaphore waiting and return the card, then just call it from `getRandomCard/getCard(fuzzy:)/getCard(exact:)`

```swift
func synchronouslyGetCard(call: String) {
  // ...
  return card
}

func getRandomCard() {
  return synchronouslyGetCard(call: "cards/random")
}
func getCard(fuzzy: String) {
  let encoded = // ... encode
  return synchronouslyGetCard(call: "cards/named?fuzzy=\(encoded)")
}

func getCard(exact: String) {
  let encoded = // ... encode
  return synchronouslyGetCard(call: "cards/named?exact=\(encoded)")
}
```